src/hotspot/share/gc/shenandoah/heuristics/shenandoahHeuristics.cpp

```diff
@@ -212,6 +224,12 @@ void ShenandoahHeuristics::adjust_penalty(intx step) {
   assert(0 <= _gc_time_penalties && _gc_time_penalties <= 100,
          "In range before adjustment: %zd", _gc_time_penalties);
 
+  if ((_previous_trigger_declinations < 16) && (step > 0)) {
```

Comment:
Could we make this `16` be a symbolic constant?

Reply:
Thanks.  Done.